organizing_hub/signals.py

```diff
@@ -91,6 +94,31 @@ def sync_group_leader_affiliation_for_user(user):
         )
 
 
+@receiver(post_save, sender=ContactList)
+def contact_list_post_save_handler(instance, **kwargs):
+    """
+    Contact List post-save handler
+
+    Start task to send approved event promotion if contact list is complete and
+    list is not empty
+    """
+
+    contact_list = instance
+    if contact_list.status == ContactListStatus.complete.value[0] and (
+        contact_list.contacts.count() > 0
+    ):
+
+        """Check if contact list is attached to approved event promotion"""
+        if hasattr(contact_list, 'eventpromotion'):
+            event_promotion = contact_list.eventpromotion
+
+            """TODO: debug why this condition is not working as expected"""
```

Comment:
Can you expand here? What's not working as expected?

Reply:
Sorry for confusion. The if statement appears to still let through use cases that shouldn't go through. It's a minor issue because the next method verifies the same thing again, so ultimately it's not impacting results. But I don't quite understand why this if statement doesn't work right, seems to be checking stale data perhaps. Might need to do a fresh lookup of the data from db to get the correct info?

Reply:
Interesting. Okay thanks for explanation!